ui/app/src/components/workspaces/ResourceOperationsList.tsx

```diff
@@ -0,0 +1,92 @@
+import { IStackStyles, MessageBar, MessageBarType, Spinner, SpinnerSize, Stack } from "@fluentui/react";
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { HttpMethod, useAuthApiCall } from '../../useAuthApiCall';
+import { Operation } from '../../models/operation';
+import { Resource } from '../../models/resource';
+import { ApiEndpoint } from '../../models/apiEndpoints';
+import { ResourceOperationListItem } from './ResourceOperationListItem';
+import config from '../../config.json';
+
+interface ResourceOperationsListProps {
+    resource: Resource
+}   
+
+export const ResourceOperationsList: React.FunctionComponent<ResourceOperationsListProps> = (props: ResourceOperationsListProps) => {
+    const apiCall = useAuthApiCall();
+    // const { workspaceId } = useParams();
+    const { resourceId } = useParams();
+    const [resourceOperations, setResourceOperations] = useState([] as Array<Operation>)
+    const [loadingState, setLoadingState] = useState('loading');
+
+    useEffect(() => {
+        const getOperations = async () => {
+            try {
+                let ws = props.resource && props.resource.id ?
+                props.resource :
+                (await apiCall(`/${props.resource.resourcePath}`, HttpMethod.Get)).workspace;
+                
+                // get resource operations 
+                const resourceOperations = await apiCall(`${ApiEndpoint.Workspaces}/${ws.id}/${ApiEndpoint.Operations}`, HttpMethod.Get, ws.properties.app_id);
```

Comment:
instead of `ws.properties.app_id` you should be able to the workspace context - like this:

With the other declarations (after `const apiCall = ...` ) add:
`const workspaceCtx = useContext(WorkspaceContext);`

Then when calling the operations, use `workspaceCtx.workspaceClientId` instead of `ws.properties`.

Reply:
Thanks